### ISSUE TITLE
YAMLDataset kedro-viz preview

### ISSUE THREAD

**Issue by derluke** [kedro-org/kedro-plugins#716]
## Description
YAMLDatasets are currently not supported in kedro-viz preview - it would be nice to have it.

## Context
YAML is a popular alternative to json, and used extensively throughout kedro.

## Possible Implementation
a trivial addition of 
```
    def preview(self) -> JSONPreview:
        """
        Generate a preview of the JSON dataset with a specified number of items.

        Returns:
            A string representing the JSON data for previewing.
        """
        data = self._load()

        return JSONPreview(json.dumps(data))
```
to kedro_datasets/yaml/yaml_dataset.py (possibly with a try/except around the dump - in case there are any special cases that yaml can do, but json.dumps can't.

**Comment by datajoely** [comment]
Good push! If you fancy opening a PR this can get released quickly

**Comment by derluke** [comment]
@datajoely I tried, and failed. The PR is red, but I don't understand why

**Comment by derluke** [comment]
https://github.com/kedro-org/kedro-plugins/pull/718

**Comment by DimedS** [comment]
@derluke many thanks for the PR! I re-runned some tests, all right now. I will review and ask teammembers to do it also

**Comment by ankatiyar** [comment]
Closed by #718
